homeassistant/components/octoprint/button.py

```diff
@@ -124,5 +127,71 @@ async def async_press(self) -> None:
             await self.client.cancel_job()
 
 
+class OctoprintShutdownSystemButton(OctoprintButton):
+    """Shutdown the system."""
+
+    def __init__(
+        self,
+        coordinator: OctoprintDataUpdateCoordinator,
+        device_id: str,
+        client: OctoprintClient,
+    ) -> None:
+        """Initialize a new OctoPrint button."""
+        super().__init__(coordinator, "Shutdown System", device_id, client)
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self.client.shutdown()
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return self.coordinator.last_update_success
+
+
+class OctoprintRebootSystemButton(OctoprintButton):
+    """Reboot the system."""
+
+    def __init__(
+        self,
+        coordinator: OctoprintDataUpdateCoordinator,
+        device_id: str,
+        client: OctoprintClient,
+    ) -> None:
+        """Initialize a new OctoPrint button."""
+        super().__init__(coordinator, "Reboot System", device_id, client)
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self.client.reboot_system()
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return self.coordinator.last_update_success
+
+
+class OctoprintRestartOctoprintButton(OctoprintButton):
```

Comment:
These both should get the RESTART device class. I'm not sure if anything gets weird if two entities have restart device class, so you should test it